tests/tuples.rs

```diff
@@ -61,6 +61,30 @@ fn tuple_windows() {
     assert_eq!(None, iter.next());
 }
 
+#[test]
+fn circular_tuple_windows() {
+    let v = [1, 2, 3, 4, 5];
+
+    let mut iter = v.iter().cloned().circular_tuple_windows();
+    assert_eq!(Some((1,)), iter.next());
+    assert_eq!(Some((2,)), iter.next());
+    assert_eq!(Some((3,)), iter.next());
+
+    let mut iter = v.iter().cloned().circular_tuple_windows();
+    assert_eq!(Some((1, 2, 3, 4)), iter.next());
+    assert_eq!(Some((2, 3, 4, 5)), iter.next());
+    // Testing impl Clone for CircularTupleWindows
+    let mut iter2 = iter.clone();
+    assert_eq!(Some((3, 4, 5, 1)), iter.next());
+    assert_eq!(Some((3, 4, 5, 1)), iter2.next());
+    assert_eq!(Some((4, 5, 1, 2)), iter.next());
+    assert_eq!(Some((4, 5, 1, 2)), iter2.next());
+    assert_eq!(Some((5, 1, 2, 3)), iter.next());
+    assert_eq!(Some((5, 1, 2, 3)), iter2.next());
+    assert_eq!(None, iter.next());
+    assert_eq!(None, iter2.next());
+}
+
```

Comment:
Could you make this a quickcheck test instead? You can use [`assert_equal`](https://docs.rs/itertools/latest/itertools/fn.assert_equal.html) to assert that both iterators produce identical results.

Reply:
Sure!